pkg/oomstore/snapshot.go

```diff
@@ -0,0 +1,46 @@
+package oomstore
+
+import (
+	"context"
+	"sort"
+
+	"github.com/oom-ai/oomstore/internal/database/metadata/sqlutil"
+
+	"github.com/oom-ai/oomstore/internal/database/metadata"
+
+	"github.com/oom-ai/oomstore/internal/database/offline"
+)
+
+func (s *OomStore) Snapshot(ctx context.Context, groupName string) error {
+	group, err := s.metadata.GetGroupByName(ctx, groupName)
+	if err != nil {
+		return err
+	}
+	revisions, err := s.metadata.ListRevision(ctx, &group.ID)
+	if err != nil {
+		return err
+	}
+	sort.Slice(revisions, func(i, j int) bool {
+		return revisions[i].Revision < revisions[j].Revision
+	})
+	for i, revision := range revisions {
+		if revision.SnapshotTable != "" {
+			continue
+		}
```

Comment:
I see the potential of turning these code into

```
SELECT * FROM revisions where snapshot_table IS NOT NULL ORDER BY revision
```

But we can do it later when we need better performance. so far so good

Reply:
but we still need the first one to be not null